### ISSUE TITLE
Godot 4: Blocky TextureLOD with OpenGL

### ISSUE THREAD

**Issue by vinrato** [godotengine/godot#72823]
### Godot version

v4.0.beta17.official [c40020513]

### System information

Windows 11, OpenGL (Compatibility), RTX 2060 (31.0.15.2824)

### Issue description

I tried to make a blurred panel for a ui element with texturelod via shader:

```glsl
shader_type canvas_item;

uniform sampler2D SCREEN_TEXTURE : hint_screen_texture, filter_linear_mipmap_anisotropic;
uniform float blur_amount : hint_range(0.0, 5.0);

void fragment() {
	vec4 blurred = textureLod(SCREEN_TEXTURE, SCREEN_UV, blur_amount);
	COLOR = blurred;
}
```

but what I got instead is this:
![image](https://user-images.githubusercontent.com/72286622/217170426-49f6712d-47ef-4c53-b9f1-30d8ad868084.png)

vulkan renderer does not exhibit this issue:
![image](https://user-images.githubusercontent.com/72286622/217170565-a013c712-94f6-49d1-b9ec-091c28a7790b.png)

### Steps to reproduce

1. Add a TextureRect node
2. Add a Panel node to overlay the previously added node
3. Apply the blur shader to the panel node

### Minimal reproduction project

N/A

**Comment by Calinou** [comment]
This is because the mipmap roughness generator is not used in the OpenGL renderer (most likely for performance reasons).

**Comment by clayjohn** [comment]
An option for Gaussian blurred mipmaps seems like a good idea. The reason they weren't used in the OpenGL renderer is that they are super slow to generate on mobile and integrated GPUs. 

That being said, now that generating mipmaps for SCREEN_TEXTURE is entirely optional, users get to choose whether to take that performance hit. And there probably aren't many uses for non-Gaussian mipmaps in 2D.

**Comment by lostminds** [comment]
> That being said, now that generating mipmaps for SCREEN_TEXTURE is entirely optional, users get to choose whether to take that performance hit. And there probably aren't many uses for non-Gaussian mipmaps in 2D.

I think this would be a good idea as a very common use of this will be to blur the background when pausing a game or opening a menu of some sort. In these cases performance will not be critical, but getting the nicer blur appearance would be very good.

**Comment by Calinou** [comment]
> I think this would be a good idea as a very common use of this will be to blur the background when pausing a game or opening a menu of some sort. In these cases performance will not be critical, but getting the nicer blur appearance would be very good.

This can be done by setting the Environment background mode to Canvas, enabling Glow then configuring it to act as a blur shader: https://github.com/godotengine/godot-proposals/issues/3485#issuecomment-955574618

**Comment by lostminds** [comment]
> This can be done by setting the Environment background mode to Canvas, enabling Glow then configuring it to act as a blur shader: [godotengine/godot-proposals#3485 (comment)](https://github.com/godotengine/godot-proposals/issues/3485#issuecomment-955574618)

Yes, this is a great technique! I used it (based on your linked tip) for my game to get a [nice background blur on the mobile renderer](https://lostminds.com/lib/style/images/screenshots/Vlobs_1.0.0_title_gamemode.png) since that supports Glow (but not textureLod I think?). However, as you noted Environment Glow isn't supported on Compatibility yet and there are cases where you might not want to do a full screen blur like the Environment Glow technique entails.

**Comment by djrain** [comment]
Agree that having the option for gaussian mipmaps would be nice. We were forced back to compatibility mode for performance reasons (I suspect due to the lack of batching on vulkan). But it's a bit of a bummer losing the nice blur and I'm not sure there's a good workaround for our use case.
